api/v1beta1/machine_types.go

```diff
@@ -230,10 +230,22 @@ type MachineStatus struct {
 	// Conditions defines current service state of the Machine.
 	// +optional
 	Conditions Conditions `json:"conditions,omitempty"`
+
+	// Deletion is the deletion state of the Machine.
+	// +optional
+	Deletion MachineStatusDeletion `json:"deletion,omitempty"`
 }
 
 // ANCHOR_END: MachineStatus
 
+// MachineStatusDeletion is the deletion state of the Machine.
+type MachineStatusDeletion struct {
```

Comment:
Should we include information here related to the drain, such as the configuration for the timeout?

Since the drain fields are optional in the spec, it would be good perhaps to show the configured values here so that you can correlate between start time and expected end time just by looking at the status?

Reply:
This is an interesting idea, not really sure how we can represent we are waiting forever in a clear way (not showing timeout 0).

Reply:
A value of `-1` potentially? But you're right, timeout 0 is awkward 🤔

Reply:
I think the status that it is still waiting should then be part of the condition `Deleting` condition message? 🤔 (or as long as that one is not around, the `DrainSucceeded` condition message.

Reply:
Yep, having this in the conditions seems the easiest way to address this